src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -65,34 +68,30 @@ public override async Task ProcessAsync(Memory<byte> data)
         {
             Stopwatch stopwatch = Stopwatch.StartNew();
             IncrementBytesReceivedMetric(data.Length);
-            var responses = await _jsonRpcProcessor.ProcessAsync(Encoding.UTF8.GetString(data.Span), _jsonRpcContext);
-
-            if (responses.Count == 1 && !responses[0].IsCollection)
-            {
-                using var result = responses[0];
-                var size = await SendJsonRpcResult(result);
-                long handlingTimeMicroseconds = stopwatch.ElapsedMicroseconds();
-                
-                _jsonRpcLocalStats.ReportCall(result.Report, handlingTimeMicroseconds, size);
-                
-                IncrementBytesSentMetric(size);
-            }
-            else if (responses.Count > 0)   // multiple responses or one array response
+            if (MemoryMarshal.TryGetArray(data, out ArraySegment<byte> segment))
             {
+                using TextReader request = new StreamReader(new MemoryStream(segment.Array!), Encoding.UTF8);
```

Comment:
Can we use COuntingTextReader here?

Reply:
No need, we already read to the Memory<byte>, so we can just use Lenght there